From 5d141d39175b16155ac5a7f5e840e2f199601c0a Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Mon, 3 Jan 2022 01:01:03 +0100
Subject: [PATCH] kernel: Add BOARD_CUSTOM_KERNEL_MK support

Allow KERNEL_OUT to be overriden for eg. building embedded uncompressed
ramdisk before building the kernel.

Change-Id: I06ad3f1eed1bf174bb6fb294388c2912e0a590d8
Signed-off-by: Dominggoes Isakh <drjisakh@gmail.com>
---
 build/tasks/kernel.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
index af495d99..0e1b2eda 100644
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -415,8 +415,14 @@ KERNEL_RECOVERY_MODULES_OUT := $(TARGET_RECOVERY_ROOT_OUT)
 $(recovery_uncompressed_ramdisk): $(TARGET_PREBUILT_INT_KERNEL)
 endif
 
+ifdef BOARD_CUSTOM_KERNEL_MK
+include $(BOARD_CUSTOM_KERNEL_MK)
+endif
+
+ifndef BOARD_CUSTOM_KERNEL_MK
 $(KERNEL_OUT):
 	mkdir -p $(KERNEL_OUT)
+endif
 
 $(KERNEL_CONFIG): $(KERNEL_OUT) $(ALL_KERNEL_DEFCONFIG_SRCS)
 	@echo "Building Kernel Config"
-- 
2.37.2

